modules/terraform/count.go

```diff
@@ -0,0 +1,90 @@
+package terraform
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// ResourceCount represents counts of resources affected by terraform apply/plan/destroy command.
+type ResourceCount struct {
+	Add     int
+	Change  int
+	Destroy int
+}
+
+// Regular expressions for terraform commands stdout pattern matching.
+const (
+	applyRegexp             = `Apply complete! Resources: (\d+) added, (\d+) changed, (\d+) destroyed\.`
+	destroyRegexp           = `Destroy complete! Resources: (\d+) destroyed\.`
+	planWithChangesRegexp   = `(\033\[1m)?Plan:(\033\[0m)? (\d+) to add, (\d+) to change, (\d+) to destroy\.`
+	planWithNoChangesRegexp = `No changes\. Infrastructure is up-to-date\.`
+)
+
+const getResourceCountErrMessage = "Can't parse Terraform output"
+
+// GetResourceCount parses stdout/stderr of apply/plan/destroy commands and returns number of affected resources.
+// This will fail the test if given stdout/stderr isn't a valid output of apply/plan/destroy.
+func GetResourceCount(t *testing.T, stdout string) *ResourceCount {
```

Comment:
NIT: Can you rename `stdout` to `cmdout`? This makes it more consistent with the function comment (since this is supposed to take in `stdout + stderr`).

Reply:
done